src/parser/rules.rs

```diff
@@ -169,6 +181,15 @@ impl Parser {
         })
     }
 
+    fn parse_inline_function(&mut self) -> Result<Statement, String> {
+        self.next()?; 
+        let expr = self.parse_expression()?;
+        let return_statment = Statement::Return(Some(expr));
+        let statements = vec![return_statment];
+        let scope = vec![];
+        Ok(Statement::Block { statements, scope })
```

Comment:
This could probably use non-shorthand field initialization syntax:

```rust
Ok(Statement::Block {
    statements: vec![return_statement],
    scope: vec![]
})
```